pkg/kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go

```diff
@@ -208,24 +211,35 @@ type muxRangeFeedEventReceiver interface {
 // startSingleRangeFeed looks up routing information for the
 // span, and begins execution of rangefeed.
 func (m *rangefeedMuxer) startSingleRangeFeed(
-	ctx context.Context, rs roachpb.RSpan, startAfter hlc.Timestamp, token rangecache.EvictionToken,
+	ctx context.Context,
+	rs roachpb.RSpan,
+	startAfter hlc.Timestamp,
+	token rangecache.EvictionToken,
+	fromManualSplit bool,
 ) error {
 	// Bound the partial rangefeed to the partial span.
 	span := rs.AsRawSpanWithNoLocals()
 
 	// Register active mux range feed.
 	stream := &activeMuxRangeFeed{
-		activeRangeFeed: newActiveRangeFeed(span, startAfter, m.registry, m.metrics),
+		// TODO(msbutler): it's sad that theres a bunch of repeat metadata.
+		// Deduplicate once old style rangefeed code is banished from the codebase.
+		activeRangeFeed: newActiveRangeFeed(span, startAfter, m.registry, m.metrics, fromManualSplit),
 		rSpan:           rs,
 		startAfter:      startAfter,
 		token:           token,
+		fromManualSplit: fromManualSplit,
 	}
 
 	if err := stream.start(ctx, m); err != nil {
 		stream.release()
 		return err
 	}
 
+	if m.cfg.withMetadata {
+		sendMetadata(m.eventCh, span, fromManualSplit)
```

Comment:
Since we are always sending metadata on startup, part of me wonders if the thing to flow through wasn't just this fromManualSplit boolean but the entire restart reason.  But, I suppose that would make the logic around the RangeKeyMismatchError a bit more complicated. I'd perhaps leave it for now, but just some food for thought.

Reply:
Another note that doesn't require a change: One note is that this changes this function from one that doesn't appear to be subject to any blocking on eventCh to one that is.

Reply:
yeah, if somebody wants to add more metadata for their use case or speed this up via some async sending, I think they should. But i didn't want to over design initially.